src/main/java/executor/service/service/proxy/impl/ProxySourcesClientImpl.java

```diff
@@ -0,0 +1,117 @@
+package executor.service.service.proxy.impl;
+
+import executor.service.handler.ProxyHandler;
+import executor.service.handler.TerminatorListener;
+import executor.service.model.proxy.ProxyConfigHolder;
+import executor.service.service.proxy.ProxyProvider;
+import executor.service.service.proxy.ProxySourcesClient;
+import executor.service.service.proxy.ProxyValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+/**
+ * The {@code ProxySourcesClientImpl} class is an implementation of the {@link ProxySourcesClient}
+ * interface. It is responsible for executing the retrieval and handling of proxy configurations
+ * from a {@link ProxyProvider}. Proxies are continuously processed in a loop with a specified delay.
+ * <p>
+ * The proxy configurations are retrieved using the specified provider and validated using the validator.
+ * Valid proxies are then processed using the provided handler.
+ * <p>
+ *
+ * @author Yurii Kotsiuba
+ * @see ProxySourcesClient
+ * @see ProxyProvider
+ * @see ProxyValidator
+ * @see ProxyConfigHolder
+ */
+public class ProxySourcesClientImpl implements ProxySourcesClient {
+
+    private static final Logger log = LoggerFactory.getLogger(ProxySourcesClient.class);
+    private static final long DELAY = 1000;
```

Comment:
In your code, the delay between iterations is set to 1000 milliseconds (1 second) and cannot be changed. Consider adding a configuration that allows users to set their own delay if they so desire.